### PR TITLE
Fix error messages; add tests

### PR DESCRIPTION
**Pull request by lavalamp**
@vishh found some super unhelpful error messages in conversion; this fixes that, adding introspectable error messages.

I also added a test, which caused me to find a bug.

### CODE REVIEW

File: pkg/api/helpers.go [hunk 1]
```diff
@@ -17,12 +17,29 @@ limitations under the License.
...
+type ConversionError struct {
+	In, Out interface{}
+	Message string
+}
+
+// Return a helpful string about the error
+func (c *ConversionError) Error() string {
+	return spew.Sprintf(
+		"Conversion error: %s. (in: %v(%+v) out: %v)",
+		c.Message, reflect.TypeOf(c.In), c.In, reflect.TypeOf(c.Out),
```
Comment (thockin) [inline pkg/api/helpers.go]: Can't you just say %T rather than using reflect?

File: pkg/api/helpers.go [hunk 2]
```diff
@@ -38,8 +55,12 @@ var Semantic = conversion.EqualitiesOrDie(
 		if b.Amount == nil && a.MilliValue() == 0 {
 			return true
 		}
+		if a.Amount == nil || b.Amount == nil {
```
Comment (thockin) [inline pkg/api/helpers.go]: Can they both be nil?

Comment (lavalamp) [inline pkg/api/helpers.go]: It only takes one nil to cause a crash on the next line...

File: pkg/api/helpers.go (hunk 1 above)
Comment (lavalamp) [inline pkg/api/helpers.go]: Cool, didn't know that one--doesn't seem to work with spew, though.

File: pkg/api/helpers.go (hunk 2 above)
Comment (lavalamp) [inline pkg/api/helpers.go]: It's subtle-- if both are nil, the first if will catch it; MilliValue() is safe to call and returns zero in that case.

Comment (thockin) [inline pkg/api/helpers.go]: Is it legit to call a method on a nil object?

Comment (lavalamp) [inline pkg/api/helpers.go]: Note that a & b can't be nil. I made quantities safe to use in an uninitialized state.

**Pull request merged**